share/functions/ls.fish

```diff
@@ -24,11 +24,17 @@ function ls --description "List contents of directory"
     # BSD, macOS and others support colors with ls -G.
     # GNU ls and FreeBSD ls takes --color=auto. Order of this test is important because ls also takes -G but it has a different meaning.
     # Solaris 11's ls command takes a --color flag.
+    # OpenBSD requires the separate colorls program
     # Also test -F because we'll want to define this function even with an ls that can't do colors (like NetBSD).
-    if not set -q __fish_ls_color_opt
+    if not set -q __fish_ls_command
+        if which colorls >/dev/null 2>/dev/null
```

Comment:
Please don't use `which`. `which` is an external tool that has a variety of issues.

Use `command -sq`, which is builtin.

Reply:
Also, honestly, there is no reason this needs to be cached in a cheesy global variable.

Just either check it right when you're about to run ls, or define a new function when `colorls` exists.

Reply:
Great, thank you for the feedback! I'll take another shot at this incorporating your feedback soon.